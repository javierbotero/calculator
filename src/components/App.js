import React, { Component } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';

export default class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      result: '0',
    };
  }

  render() {
    const { result } = this.state;
    return (
      <>
        <Display result={result} />
        <ButtonPanel />
      </>
    );
  }
}
