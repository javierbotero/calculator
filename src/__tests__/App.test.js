import React from 'react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import {
  render, fireEvent, waitFor, screen, getByText
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from '../components/App';

describe('Testing App', () => {
  it('Shows text "Let\'s do some math!"', () => {
    render(<App />);
    expect(screen.getByText('Let\'s do some math!').innerHTML).toBe('Let\'s do some math!');
  });

  it('Press button one and display shows 1', () => {
    render(<App />);
    const one = screen.getByText('1');
    fireEvent.click(one);
    const display = document.querySelector('.display');
    expect(display).toHaveTextContent('1');
  });

  it('Makes an operation and displays the resul', () => {
    render(<App />);
    fireEvent.click(screen.getByText('9'));
    fireEvent.click(screen.getByText('0'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('='));
    const display = document.querySelector('.display');
    expect(display).toHaveTextContent('180');
  });
  
  it('Checks that display shows 0 if AC is pressed after some buttons are pressed', () => {
    render(<App />);
    fireEvent.click(screen.getByText('8'));
    fireEvent.click(screen.getByText('7'));
    const display = document.querySelector('.display');
    expect(display).toHaveTextContent('87');
    fireEvent.click(screen.getByText('AC'));
    expect(display).toHaveTextContent('0');
  });
});
