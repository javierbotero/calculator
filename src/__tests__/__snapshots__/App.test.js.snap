// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Makes sure App is the same with the snapshot 1`] = `
Array [
  <h3>
    Let's do some math!
  </h3>,
  <div
    className="parent"
  >
    <div
      className="display"
    >
      0
    </div>
    <div
      className="container"
    >
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        AC
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        +/-
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        %
      </div>
      <div
        className="key keyWidthGeneral #f5933d"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f5933d",
            "flexBasis": "25%",
          }
        }
      >
        /
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        7
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        8
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        9
      </div>
      <div
        className="key keyWidthGeneral #f5933d"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f5933d",
            "flexBasis": "25%",
          }
        }
      >
        x
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        4
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        5
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        6
      </div>
      <div
        className="key keyWidthGeneral #f5933d"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f5933d",
            "flexBasis": "25%",
          }
        }
      >
        -
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        1
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        2
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        3
      </div>
      <div
        className="key keyWidthGeneral #f5933d"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f5933d",
            "flexBasis": "25%",
          }
        }
      >
        +
      </div>
      <div
        className="key keyWidthSpecial #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "50%",
          }
        }
      >
        0
      </div>
      <div
        className="key keyWidthGeneral #e1e1e1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#e1e1e1",
            "flexBasis": "25%",
          }
        }
      >
        .
      </div>
      <div
        className="key keyWidthGeneral #f5933d"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f5933d",
            "flexBasis": "25%",
          }
        }
      >
        =
      </div>
    </div>
  </div>,
]
`;
